File: playwright/check-website.js
const { firefox } = require("playwright");

const HEADLESS = process.env.HEADLESS === 'true'; // toggle via env variable

function logNote(message) {
  const timestamp = String(BigInt(Date.now()) * 1000000n).slice(0, 16);
  console.log(`${timestamp} ${message}`);
}

async function main() {
  logNote("Launch browser Firefox");
  const browser = await firefox.launch({ headless: HEADLESS });

  logNote("Creating new context");
  const context = await browser.newContext({
      ignoreHTTPSErrors: true,
      viewport: { width: 1280, height: 720 },
  });

  logNote("Creating new page");
  const page = await context.newPage();

  logNote("Visiting http://ngnix");

  const response = await page.goto('http://ngnix', {
    waitUntil: 'domcontentloaded',
    timeout: 5000,
  });

  if (!response || !response.ok()) {
    throw new Error(`Unexpected response: ${response ? response.status() : "no response"}`);
  }

  logNote("Idling for 5s");
  await page.waitForTimeout(5000);

  logNote("Closing Browser");
  await browser.close();

  logNote("End");
}

main().catch((error) => {
  console.error(error);
  process.exit(1);
});

File: playwright/check-website.py
from playwright.sync_api import sync_playwright
import sys
import os
from time import time_ns

HEADLESS = os.getenv("HEADLESS", "false").lower() == "true"

def log_note(message: str) -> None:
    timestamp = str(time_ns())[:16]
    print(f"{timestamp} {message}")

def main():

    with sync_playwright() as p:
        log_note("Launch browser Firefox")
        browser = p.firefox.launch(headless=HEADLESS)

        log_note("Creating new context")
        context = browser.new_context(ignore_https_errors=True, viewport={'width': 1280, 'height': 720})

        log_note("creating new page")
        page = context.new_page()

        log_note("Visiting http://ngnix")

        response = page.goto(
            "http://ngnix",
            wait_until="domcontentloaded",
            timeout=5000
        )

        if response is None or not response.ok:
            status = response.status if response else "no response"
            raise Exception(f"Unexpected response: {status}")

        log_note("Idling for 5s")

        page.wait_for_timeout(5000)

        log_note("Closing Browser")
        browser.close()

        log_note("End")

if __name__ == "__main__":
    try:
        main()
    except Exception as e:
        print(e)
        sys.exit(1)